standard_sequence/experiment_components.py

```diff
@@ -0,0 +1,816 @@
+        devices.uv_switch.go_high(t)
+        t += dur
+        devices.uv_switch.go_low(t)
+        return t
+
+
+class BField:
+    CONST_COIL_OFF_TIME: ClassVar[float] = (
+        1.4e-3  # minimum time for the MOT coil to be off
+    )
+    CONST_COIL_RAMP_TIME: ClassVar[float] = (
+        100e-6  # ramp time from initial field to final field
+    )
+    CONST_BIPOLAR_COIL_FLIP_TIME: ClassVar[float] = (
+        10e-3  # the time takes to flip the polarity of the coil
+    )
+    CONST_COIL_FEEDBACK_OFF_TIME: ClassVar[float] = (
+        4.5e-3  # how long to turn off the feedback of circuit when flipping polarity
+    )
+
+    bias_voltages = list[float]
+    mot_coils_on: bool
+    mot_coils_on_current: float
+    current_outputs = tuple[AnalogOut, AnalogOut, AnalogOut]
+    feedback_disable_ttls = tuple[DigitalOut, DigitalOut, DigitalOut]
+
+    def __init__(self, t: float):
+        self.bias_voltages = [
+            shot_globals.mot_x_coil_voltage,
+            shot_globals.mot_y_coil_voltage,
+            shot_globals.mot_z_coil_voltage,
+        ]
+        self.mot_coils_on = shot_globals.do_mot_coil
+        self.mot_coils_on_current = 10 / 6
+
+        # Same question as for Dline_lasers, should we automatically initialize the hardware here or in a separate function we can call?
+
+        # initialize bias_field variable to None
+        # self.bias_field = None
+
+        self.t_last_change = 0
+
+        self.current_outputs = (
+            devices.x_coil_current,
+            devices.y_coil_current,
+            devices.z_coil_current,
+        )
+
+        self.feedback_disable_ttls = (
+            devices.x_coil_feedback_off,
+            devices.y_coil_feedback_off,
+            devices.z_coil_feedback_off,
+        )
+
+        for current_output, bias_voltage_cmpnt in zip(
+            self.current_outputs, self.bias_voltages
+        ):
+            current_output.constant(t, bias_voltage_cmpnt)
+
+        if self.mot_coils_on:
+            devices.mot_coil_current_ctrl.constant(t, self.mot_coils_on_current)
+        else:
+            devices.mot_coil_current_ctrl.constant(t, 0)
+
+    def flip_coil_polarity(
+        self, t: float, final_voltage: float, component: Literal[0, 1, 2]
+    ):
+        """
+        t: float
+            Time to begin coil polarity flipping
+        final_voltage: float
+            Final control voltage on the coil
+        component: {0, 1, 2}
+            Specifies the field component to flip (x, y, or z)
+        """
+        coil_voltage_mid_abs = 0.03
+        coil_voltage_mid = np.sign(final_voltage) * coil_voltage_mid_abs
+        total_coil_flip_ramp_time = (
+            self.CONST_BIPOLAR_COIL_FLIP_TIME + self.CONST_COIL_RAMP_TIME
+        )
+
+        current_output = self.current_outputs[component]
+        feedback_disable_ttl = self.feedback_disable_ttls[component]
+
+        t += current_output.ramp(
+            t,
+            duration=self.CONST_COIL_RAMP_TIME / 2,

```

Comment:
if this const is divided by 2, should the const be redefined?

Reply:
No, if the polarity is not flipped, the total ramp will be const. It is only divided by 2 when flipping polarity